components/identity/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/clientauth/AbstractClientAuthHandler.java

```diff
@@ -0,0 +1,88 @@
+*software distributed under the License is distributed on an
+*"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*KIND, either express or implied.  See the License for the
+*specific language governing permissions and limitations
+*under the License.
+*/
+
+package org.wso2.carbon.identity.oauth2.token.handlers.clientauth;
+
+import org.apache.axis2.util.JavaUtils;
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AccessTokenReqDTO;
+import org.wso2.carbon.identity.oauth2.token.OAuthTokenReqMessageContext;
+
+import java.util.Properties;
+
+public abstract class AbstractClientAuthHandler implements ClientAuthenticationHandler {
+
+    Properties properties;
+
+    public boolean canAuthenticate(OAuthTokenReqMessageContext tokReqMsgCtx)
+            throws IdentityOAuth2Exception {
+
+        OAuth2AccessTokenReqDTO oAuth2AccessTokenReqDTO = tokReqMsgCtx.getOauth2AccessTokenReqDTO();
+
+        if (StringUtils.isNotEmpty(oAuth2AccessTokenReqDTO.getClientId()) &&
+                StringUtils.isNotEmpty(oAuth2AccessTokenReqDTO.getClientSecret())) {
+            return true;
+
+        } else {
+            if (org.wso2.carbon.identity.oauth.common.GrantType.SAML20_BEARER.toString().equals(
+                    oAuth2AccessTokenReqDTO.getGrantType())) {
+
+                Properties confProperties = getProperties();
+                String authConfig = confProperties.getProperty(
+                        OAuthConstants.CLIENT_AUTH_CREDENTIAL_VALIDATION);
+
+                //If user has set strict validation to false, can authenticate without credentials
+                if (StringUtils.isNotEmpty(authConfig) && JavaUtils.isFalseExplicitly(authConfig)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public boolean authenticateClient(OAuthTokenReqMessageContext tokReqMsgCtx)
+            throws IdentityOAuth2Exception {
+
+        OAuth2AccessTokenReqDTO oAuth2AccessTokenReqDTO = tokReqMsgCtx.getOauth2AccessTokenReqDTO();
+
+        if (StringUtils.isEmpty(oAuth2AccessTokenReqDTO.getClientSecret())) {
+
+            //Getting configured value for client credential validation requirements
+            String authConf = getProperties().getProperty(
+                    OAuthConstants.CLIENT_AUTH_CREDENTIAL_VALIDATION);
+
+            //Skipping credential validation for saml2 bearer if not configured as needed
+            if (org.wso2.carbon.identity.oauth.common.GrantType.SAML20_BEARER.toString()
+                    .equals(oAuth2AccessTokenReqDTO.getGrantType()) &&
+                    JavaUtils.isFalseExplicitly(authConf)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public Properties getProperties() {
```

Comment:
No need getter/setter. Super class has a init(Properties) method. Properties should get set from that.